File: byteback-test/test-scala-2.13.8/src/main/scala/byteback/test/exceptions/Basic.scala
/**
 * RUN: %{byteback} -cp %{jar} -c %{class} -o %t.bpl
 */
package byteback.test.exceptions;

import byteback.specification.Contract._;
import byteback.specification.Special._;
import byteback.specification.Operators._;
import byteback.specification.Operators.{eq => equal};

import scala.annotation.meta._;

class Basic(@(Behavior @getter) val f: Int) {

  def tryCatchBlock(): Unit = {
    try {
      throw new Exception();
    } catch {
      case _: Exception =>
    }
  }

  @Return
  def neverThrows(): Unit = {
  }

  def neverCatches(): Unit = {
    try {
      neverThrows();
    } catch {
      case _: Exception => assertion(false);
    }
  }

  @Behavior
  def always_throws(): Boolean = {
    return true;
  }

  @Raise(exception = classOf[Exception], when = "always_throws")
  def alwaysThrows(): Unit = {
    throw new Exception();
  }

  def alwaysCatches(): Unit = {
    try {
      alwaysThrows();
      assertion(false);
    } catch {
      case _: Throwable =>
    }
  }

  @Raise(exception = classOf[Exception], when = "always_throws")
  def callsAlwaysThrows(): Unit = {
    alwaysThrows();
  }

  @Behavior
  def argument_is_even(n: Int): Boolean = {
    return equal(n % 2, 0);
  }

  @Raise(exception = classOf[Exception], when = "argument_is_even")
  def throwsIfEven(n: Int): Unit = {
    if (n % 2 == 0) {
      throw new Exception();
    }
  }

  def catchesIfEven(): Unit = {
    try {
      throwsIfEven(2);
      assertion(false);
    } catch {
      case e: Exception => assertion(true);
    }
  }

  @Behavior
  def f_is_1(): Boolean = {
    return equal(f, 1);
  }

  @Behavior
  def f_is_2(): Boolean = {
    return equal(f, 2);
  }

  @Behavior
  def f_is_3(): Boolean = {
    return equal(f, 3);
  }

  @Behavior
  def f_is_4(): Boolean = {
    return equal(f, 4);
  }

  @Behavior
  def f_is_gt_4(): Boolean = {
    return gt(f, 4);
  }

  @Raise(exception = classOf[Exception1], when = "f_is_1")
  @Raise(exception = classOf[Exception2], when = "f_is_2")
  @Raise(exception = classOf[Exception3], when = "f_is_3")
  @Raise(exception = classOf[Exception4], when = "f_is_4")
  @Return(when = "f_is_gt_4")
  def throwsMultiple(): Unit = {
    f match {
      case 1 => throw new Exception1()
      case 2 => throw new Exception2()
      case 3 => throw new Exception3()
      case 4 => throw new Exception4()
      case _ =>
    }
  }

}
/**
 * RUN: %{verify} %t.bpl | filecheck %s
 * CHECK: Boogie program verifier finished with 10 verified, 0 errors
 */

File: byteback-test/test-kotlin-1.8.0/src/main/kotlin/byteback/test/exceptions/Basic.kt
/** 
 * RUN: %{byteback} -cp %{jar} -c %{class} -o %t.bpl	
 */
package byteback.test.exceptions

import byteback.specification.Contract.*
import byteback.specification.Operators.*

class Basic {

  @get:Behavior val f: Int = 0

  fun tryCatchBlock(): Exception {
    try {
      throw Exception()
    } catch (e: Exception) {
      return e
    }
  }

  @Return
	fun neverThrows(): Unit {}

  fun neverCatches(): Unit {
    try {
      neverThrows()
    } catch (e: Exception) {
      assertion(false)
    }
  }

  @Behavior
  fun `always throws`(): Boolean {
    return true
  }

  @Raise(`exception` = Exception::class, `when` = "always throws")
  fun alwaysThrows(): Unit {
    throw Exception()
  }

  fun alwaysCatches(): Unit {
    try {} catch (e: Exception) {}
  }

  fun callsAlwaysThrows(): Unit {
    try {
      alwaysThrows()
      assertion(false)
    } catch (e: Exception) {}
  }

  @Behavior
  fun `argument is even`(n: Int): Boolean {
    return eq(n % 2, 0)
  }

  @Raise(`exception` = Exception::class, `when` = "argument is even")
  fun throwsIfEven(n: Int): Unit {
    if (n % 2 == 0) {
      throw Exception()
    }
  }

  fun catchesIfEven(): Unit {
    try {
      throwsIfEven(2)
      assertion(false)
    } catch (e: Exception) {}
  }

  @Behavior
  fun `f is 1`(): Boolean {
    return eq(f, 1)
  }

  @Behavior
  fun `f is 2`(): Boolean {
    return eq(f, 2)
  }

  @Behavior
  fun `f is 3`(): Boolean {
    return eq(f, 3)
  }

  @Behavior
  fun `f is 4`(): Boolean {
    return eq(f, 4)
  }

  @Behavior
  fun `f is gt 4`(): Boolean {
    return gt(f, 4)
  }

  @Raise(`exception` = Exception1::class, `when` = "f is 1")
  @Raise(`exception` = Exception2::class, `when` = "f is 2")
  @Raise(`exception` = Exception3::class, `when` = "f is 3")
  @Raise(`exception` = Exception4::class, `when` = "f is 4")
  @Return(`when` = "f is gt 4")
  fun throwsMultiple(): Unit {
    if (f == 1) {
      throw Exception1()
    } else if (f == 2) {
      throw Exception2()
    } else if (f == 3) {
      throw Exception3()
    } else if (f == 4) {
      throw Exception4()
    }
  }
}
/**
 * RUN: %{verify} %t.bpl | filecheck %s
 * CHECK: Boogie program verifier finished with 10 verified, 0 errors
 */
